### PR TITLE
TYP: generic, series, frame

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

cc @simonjayhawkins

### CODE REVIEW

File: pandas/core/frame.py [hunk 1]
```diff
@@ -9306,8 +9306,8 @@ def _AXIS_NAMES(self) -> Dict[int, str]:
 ops.add_special_arithmetic_methods(DataFrame)
 
 
-def _from_nested_dict(data):
-    new_data = collections.defaultdict(dict)
+def _from_nested_dict(data) -> collections.defaultdict:
+    new_data: collections.defaultdict = collections.defaultdict(dict)
```
Comment (jreback) [inline pandas/core/frame.py:9310]: you *should*n't have to type this

File: pandas/core/series.py [hunk 2]
```diff
@@ -1789,12 +1789,17 @@ def count(self, level=None):
         """
         if level is None:
             return notna(self.array).sum()
+        elif not isinstance(self.index, MultiIndex):
+            raise ValueError("Series.count level is only valid with a MultiIndex")
```
Comment (jreback) [inline pandas/core/series.py:1793]: is this hit in a test?

File: pandas/core/frame.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/frame.py:9310]: i agree, but mypy complains when it isnt annotated

File: pandas/core/series.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/core/series.py:1793]: no.  ill add one in my just started collected-followups branch

**Pull request merged**